app/time_tracking/work_statistic/views.py

```diff
@@ -0,0 +1,148 @@
+from django.utils import timezone
+from datetime import timedelta, datetime
+from django.contrib.auth.models import User
+from django.db.models import Count
+
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework import status
+
+from time_tracking.work_time.models import WorkTime
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def week_statistics_detail(request, format=None):
+    """
+    Retrieve work time since a week ago.
+    """
+    if request.method == 'GET':
+        work_times = WorkTime.objects.filter(
+            owner=request.user,
+            work_time__isnull=False,
+            end_time__gt=timezone.now() - timedelta(weeks=1)
+        ).aggregate(Sum('work_time'))
+        responce = {
+            "total_working_time_in_seconds": work_times['work_time__sum']
+        }
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def quarter_statistics_detail(request, format=None):
+    """
+    Retrieve work time since a 13 weeks ago.
+    """
+    if request.method == 'GET':
+        work_times = WorkTime.objects.filter(
+            owner=request.user,
+            work_time__isnull=False,
+            end_time__gt=timezone.now() - timedelta(weeks=13)
+        ).aggregate(Sum('work_time'))
+        responce = {
+            "total_working_time_in_seconds": work_times['work_time__sum']
+        }
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def year_statistics_detail(request, format=None):
+    """
+    Retrieve work time since a 356 days ago.
+    """
+    if request.method == 'GET':
+        work_times = WorkTime.objects.filter(
+            owner=request.user,
+            work_time__isnull=False,
+            end_time__gt=timezone.now() - timedelta(days=356)
+        ).aggregate(Sum('work_time'))
+        responce = {
+            "total_working_time_in_seconds": work_times['work_time__sum']
+        }
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def employees_arrival_and_leving_time_statistics_detail(request, format=None):
+    """
+    Retrieve the employees arrival time and leaving times.
+    """
+    if request.method == 'GET':
+        responce = []
+        for user in User.objects.filter(is_staff=False):
+            avg_work_time_starts = WorkTime.objects.filter(owner=user).aggregate(Avg('start_unix_time'))['start_unix_time__avg']
+            avg_work_time_ends = WorkTime.objects.filter(owner=user).aggregate(Avg('end_unix_time'))['end_unix_time__avg']
```

Comment:
this will lead to inaccurate if the user checked/checkout multiple times in the same day

Reply:
I updated this to group the records by the date and then got the min start time, and the max end time,
and based on that I calculated the average of the start and end work-time.